jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from typing import NamedTuple, Optional, Tuple
+
+import jax.numpy as jnp
+import scipy
+from jax import jit
+from jax._src.api import vmap
+from jax._src.numpy.lax_numpy import _check_arraylike
+from jax._src.numpy.util import _wraps
+from jax._src.typing import Array, ArrayLike
+from jax._src.util import canonicalize_axis
+
+
+class ModeResult(NamedTuple):
+  mode: Array
+  count: Array
```

Comment:
If this causes mypy errors, lets switch it back to how you originally had it defined.